plutus-list/src/PlutusTx/List/Natural.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE Trustworthy #-}
+
+{- |
+ Module: PlutusTx.List.Natural
+ Copyright: (C) MLabs 2021
+ License: Apache 2.0
+ Maintainer: Koz Ross <koz@mlabs.city>
+ Portability: GHC only
+ Stability: Experimental
+ List functions that works with the 'Natural' type.
+-}
+module PlutusTx.List.Natural (
+  length,
+  replicate,
+  take,
+  drop,
+  splitAt,
+) where
+
+import Data.Monoid (Sum (Sum, getSum))
+import PlutusTx.Natural (Natural, nat)
+import PlutusTx.Prelude hiding (length, take)
+import Prelude qualified ()
+
+{-# INLINEABLE length #-}
+
+{- | Returns the size/length of a finite structure as a 'Natural'.
+
+@since 1.0
+-}
+length :: Foldable f => f a -> Natural
```

Comment:
The style guide requires [explicit foralls](https://github.com/mlabs-haskell/styleguide/blob/main/TLDR.md#type-and-kind-signatures): this is kind of necessary if we're using `ScopedTypeVariables`.